utilities/examples/md_parser.c

```diff
@@ -0,0 +1,177 @@
+    META_DATA_CAMERA_DEBUG_ID               = 0x800000FF,
+    META_DATA_HID_IMU_REPORT_ID             = 0x80001001,
+    META_DATA_HID_CUSTOM_TEMP_REPORT_ID     = 0x80001002,
+    META_DATA_MIPI_INTEL_DEPTH_ID           = 0x80010000,
+    //META_DATA_MIPI_INTEL_RGB_ID             = 0x80010001, // D457 - to be restored after the FW bug is resolved
+};
+#pragma pack(push, 1)
+/**\brief md_header - metadata header is a integral part of all rs4XX metadata objects */
+struct md_header
+{
+    enum md_type     md_type_id;         // The type of the metadata struct
+    uint32_t    md_size;            // Actual size of metadata struct without header
+};
+/**\brief md_mipi_depth_mode - properties associated with sensor configuration
+ *  during video streaming. Corresponds to FW STMetaDataExtMipiDepthIR object*/
+struct md_mipi_depth_mode
+{
+    struct md_header   header;
+    uint8_t     version;  // maybe need to replace by uint8_t for version and 3 others for reserved
+    uint16_t    calib_info;
+    uint8_t     reserved;
+    uint32_t    flags;              // Bit array to specify attributes that are valid
+    uint32_t    hw_timestamp;
+    uint32_t    optical_timestamp;   //In microsecond unit
+    uint32_t    exposure_time;      //The exposure time in microsecond unit
+    uint32_t    manual_exposure;
+    uint16_t    laser_power;
+    uint16_t    trigger;
+    uint8_t     projector_mode;
+    uint8_t     preset;
+    uint8_t     manual_gain;
+    uint8_t     auto_exposure_mode;
+    uint16_t    input_width;
+    uint16_t    input_height;
+    uint32_t    sub_preset_info;
+    uint32_t    crc;
+};
+const uint8_t md_mipi_depth_mode_size = sizeof(struct md_mipi_depth_mode);
+/**\brief md_mipi_rgb_mode - properties associated with sensor configuration
+ *  during video streaming. Corresponds to FW STMetaDataExtMipiRgb object*/
+struct md_mipi_rgb_mode
+{
+    struct md_header   header;
+    uint32_t    version;  // maybe need to replace by uint8_t for version and 3 others for reserved
+    uint32_t    flags;              // Bit array to specify attributes that are valid
+    uint32_t    hw_timestamp;
+    uint8_t     brightness;
+    uint8_t     contrast;
+    uint8_t     saturation;
+    uint8_t     sharpness;
+    uint16_t    auto_white_balance_temp;
+    uint16_t    gamma;
+    uint16_t    manual_exposure;
+    uint16_t    manual_white_balance;
+    uint8_t     auto_exposure_mode;
+    uint8_t     gain;
+    uint8_t     backlight_compensation;
+    uint8_t     hue;
+    uint8_t     power_line_frequency;
+    uint8_t     low_light_compensation;
+    uint16_t    input_width;
+    uint16_t    input_height;
+    uint8_t     reserved[6];
+    uint32_t    crc;
+};
+const uint8_t md_mipi_rgb_mode_size = sizeof(struct md_mipi_rgb_mode);
+#pragma pack(pop)
+
+/**\brief metadata_mipi_raw - metadata structure
+ *  layout as transmitted and received by backend */
+struct metadata_mipi_depth_raw
+{
+    struct uvc_header_mipi    header;
```

Comment:
No need for `#pragma pack(push, 1)` here?

Reply:
no need in specific here, but we will add it.